src/services/inplayer.account.service.ts

```diff
@@ -313,18 +314,41 @@ export const updatePersonalShelves: UpdatePersonalShelves = async (payload) => {
 
 export const exportAccountData: ExportAccountData = async () => {
   // password is sent as undefined because it is now optional on BE
-  const response = await InPlayer.Account.exportData({ password: undefined, brandingId: 0 });
-  const { code, message } = response.data;
-  if (code !== 200) {
-    throw new Error(message);
+  try {
+    const response = await InPlayer.Account.exportData({ password: undefined, brandingId: 0 });
+    const { code, message } = response.data;
+    if (code !== 200) {
+      throw new Error(message);
+    }
+    return {
+      errors: [],
+      responseData: {
+        message,
+        code,
+      },
+    };
```

Comment:
Looks like part of the code we can reuse for several endpoints using a function. You can have a look on `getDataOrThrow` as an example.